File: src/LletresRepetidesAPP.java
import java.util.*;

public class LletresRepetidesAPP {

	public static void main(String[] args) {

		// Nom amb String

		String name = "Gerard";
		String surname = "Puig";

		// Convertir el nombre en una Arraylist 

		ArrayList<Character> arraylistname = new ArrayList();

		for (int i = 0; i < name.length(); i++) {
			arraylistname.add(name.charAt(i));
		}

		// Convertir el apellido en Arraylist

		ArrayList<Character> arraylistsurname = new ArrayList();

		for (int i = 0; i < surname.length(); i++) {
			arraylistsurname.add(surname.charAt(i));
		}

		// Fusionar las dos listas con un espacio en medio

		ArrayList<Character> arraylistFullName = new ArrayList();

		for (char letra : arraylistname) {
			arraylistFullName.add(letra);
		}

		arraylistFullName.add(' ');

		for (char letra : arraylistsurname) {
			arraylistFullName.add(letra);
		}

		// Imprime el nombre completo con espacio
		
		for (char letra : arraylistFullName) {
			System.out.print(letra);
		}

		System.out.println("");

		// Imprime si son vocales o consonantes i detecta números o caracters extranys

		for (int i = 0; i < arraylistFullName.size(); i++) {

			char letra = Character.toUpperCase(arraylistFullName.get(i));

			if (letra == 'A' || letra == 'E' || letra == 'I' || letra == 'O' || letra == 'U') {

				System.out.println(arraylistFullName.get(i) + " VOCAL");

			} else if (Character.isDigit(letra)) {

				System.out.println(arraylistFullName.get(i) + " Els noms de persones no contenen números!");

			} else if (letra == ' ') {

				// no hacer nada

			} else {

				System.out.println(arraylistFullName.get(i) + " CONSONANT");

			}
		}

		// Mapear letras nombre completo

		Map<Character, Integer> contadorletras = new HashMap<Character, Integer>();

		for (char l : arraylistFullName) {

			char key = Character.toUpperCase(l);

			if (Character.isAlphabetic(key)) {

				if (contadorletras.get(key) == null) {

					contadorletras.put(key, 1);

				} else {

					contadorletras.put(key, contadorletras.get(key) + 1);

				}
			}

		}

		// Mostrar en pantalla los datos

		System.out.println(contadorletras.entrySet());		

	}
}

File: JavaScript/LletresRepetidesAPP.js
//Crear Array, añadir letras

var nombre = new Array;
var surname = new Array;

// Nom amb String

var nameinput = prompt("Escribe tu nombre:");
var surnameinput = prompt("Escribe tu apellido:");

// Convertir el nombre en una Array

for (var i = 0; i < nameinput.length; i++) {
    nombre.push(nameinput.charAt(i));
}

// Convertir el apellido en Array

for (var i = 0; i < surnameinput.length; i++) {
    surname.push(surnameinput.charAt(i));
}


// Fusionar las dos listas con un espacio en medio

var fullname = new Array;

nombre.forEach(letra =>
    fullname.push(letra));

fullname.push(' ');

surname.forEach(letra =>
    fullname.push(letra));

// Comprobar si son vocales, consonates y números
for (var i = 0; i < fullname.length; i++) {
    document.writeln(fullname[i], "<br/>");
    if (/[À-ú,a,e,i,o,u,A,E,I,O,U]/.test(fullname[i])) {
        document.writeln("VOCAL", "<br/>");

    } else if (/[1-9]/.test(fullname[i])) {
        document.writeln("Els noms de persones no contenen números!", "<br/>");
    } else if (fullname[i]==' '){
        //no hacer nada
    } else {
        document.writeln("CONSONANT", "<br/>");
    }
}

//Mapear letras nombre
		
var contadorletras = new Map;

for (let index = 0; index < fullname.length; index++) {

    if (/[A-zÀ-ú]/.test(fullname[index])){

        if (contadorletras.has(fullname[index].toUpperCase())){
            contadorletras.set(fullname[index].toUpperCase(), contadorletras.get(fullname[index].toUpperCase())+1);
        }else{
            contadorletras.set(fullname[index].toUpperCase(), 1);
        }

    }

}

contadorletras.forEach((key,value)=>{document.write(key,value,"</br>")});

